src/Storages/MergeTree/CommonANNIndexes.cpp

```diff
@@ -509,6 +521,31 @@ bool ANNCondition::matchMainParts(RPN::iterator & iter, const RPN::iterator & en
         ++iter;
     }
 
+    if (iter->function == RPNElement::FUNCTION_CAST)
```

Comment:
Just to clarify: Should the CAST parsing code be tested (using some SQL examples) or is there a test already?